### DISCUSSION TITLE
How do i maintain cache object during JsonSerialize

### DISCUSSION THREAD

**Discussion by sathyavelz** [FasterXML/jackson-databind#3922] [Q&A]
Consider, i have a pojo like this

```
public class Person {
@JsonSerialize(using = LongToNameConverter.class)
@JsonProperty(value = "first_user_id")
Long firstUserId;
@JsonSerialize(using = LongToNameConverter.class)
@JsonProperty(value = "second_user_id")
Long secondUserId;
}
```

The JSON would be

```
{
  "first_user_id": {
    "id": "123",
    "name": "userName"
  },
  "second_user_id": {
    "id": "123",
    "name": "userName"
  }
}
```

In LongToNameConverter serializer i fetch name for the person from DB, here each time i need to fecth name for the user, is it possible to cache the idVsName and can be use in the LongToNameConverter seralizer

**Comment by JooHyukKim** [comment]
@sathyavelz  You might consider using `@JsonIdentityInfo`, here is link to 🔗[javadoc](https://www.javadoc.io/doc/com.fasterxml.jackson.core/jackson-annotations/latest/com/fasterxml/jackson/annotation/JsonIdentityInfo.html) and [example usages](https://github.com/FasterXML/jackson-databind/blob/2.16/src/test/java/com/fasterxml/jackson/databind/objectid/TestObjectIdSerialization.java).

**Reply by sathyavelz** [reply]
@JooHyukKim Can i get any other way, this couldn't help.

**Reply by JooHyukKim** [reply]
@sathyavelz 
Can you explain a bit more what you want to cache, and what to use as cache key?

**Reply by sathyavelz** [reply]
@JooHyukKim  Consider i have a List of Person. and i need to convert this Obj as Json, in person i have a serializer for userId, in serializer i fetch name for the corresponding userId.

Consider the list size is 100, each tym in serializer i fetch the name, instead if i have any hasmap there i can store idVsName, in serializer i can get name from the Map . itself, if Map not has the id means i can query and set again in map, so that next time for same user i can get from map.

**Comment by cowtowncoder** [comment]
Beside what @JooHyukKim suggested, you might be able to change `Person` to have a method  that returns serialized JSON, annotated with `@JsonRawValue`; lazily creating value to return.

That annotation basically means that whatever annotated method returns will be included as-is in output. So if that is serialized JSON, that'll be inserted -- so if you construct serialization on first call, retained in `Person`, only serialized once.
